app/web/features/auth/AuthProvider.tsx

```diff
@@ -20,26 +20,26 @@ function useAppContext<T>(context: Context<T | null>) {
 export default function AuthProvider({ children }: { children: ReactNode }) {
   const store = useAuthStore();
 
-  const push = useHistory().push;
+  const router = useRouter();
```

Comment:
You're right: https://github.com/vercel/next.js/issues/18127

Reply:
Actually,  isn't this right? Because if it's not stable, we don't want to accidentally call the "old" `router.push`?

Reply:
Yeah, but then we run into that infinite loop issue I think you faced in that other `useEffect` in some cases so seems like a bug to me? It's unclear what bad things would happen if we call the "old" `router.push`... I'd assume nothing since it doesn't have router state like react-router? 🤷

Reply:
Added a workaround taken from that bug thread which allows to use the latest router.push but doesn't trigger useEffect. I also found the `t` dependency in `useAuthStore` was causing that useEffect to fire too often too, so fixed that.